middleware/message.js

```diff
@@ -0,0 +1,11 @@
+const setMessage = (req, res, next) => {
+  if (req.session.pendingMessage) {
```

Comment:
Can we refactor this if / else to a ternary expression?

Reply:
this one should be easy

```javascript
req.session.pendingMessage
    ? (res.locals.message = req.session.pendingMessage)
    : (res.locals.message = "")
```

Reply:
Hmmmmmm what is the value we're trying to change? What are we assigning a value to?

(Same question 2 different ways)

Reply:
> Hmmmmmm what is the value we're trying to change? What are we assigning a value to?
> 
> (Same question 2 different ways)

We are changing the value of the message shown to the user if there is a pending message in the session, if not we set the value to an empty string.

For example, after creating a task we set the session.pendingMessage to "Task created" and redirects the user to /tasks;
During the load we check for the session.pendingMessage, if there is a message we set the locals.message to it or we set to an empty string.

We have to do this otherwise we can't load the message to the user and the app crashes because is undefined.

Reply:
Yes but you can rewrite it in a much similar way. The _first thing_ in a ternary is the value we are _trying to assign a  value to._

```
const thingThatWillGetANewValue = condition
  ? valueIfTrue
  : valueIfFalse
```

Reply:
> Yes but you can rewrite it in a much similar way. The _first thing_ in a ternary is the value we are _trying to assign a value to._
> 
> ```
> const thingThatWillGetANewValue = condition
>   ? valueIfTrue
>   : valueIfFalse
> ```

Ohhh I didn't know I could do that! That's very cool! :grin:

Reply:
> So it would be like this:
> ```javascript
>   res.locals.message = req.session.pendingMessage
>     ? req.session.pendingMessage
>     : ""
> ```

Exactly